File: src/main/java/dev/esophose/discordbot/listener/GuildJoinLeaveListener.java
package dev.esophose.discordbot.listener;

import dev.esophose.discordbot.Sparky;
import dev.esophose.discordbot.manager.GuildSettingsManager;
import discord4j.core.event.domain.guild.GuildCreateEvent;
import discord4j.core.event.domain.guild.GuildDeleteEvent;
import discord4j.core.event.domain.guild.GuildEvent;

public class GuildJoinLeaveListener extends Listener<GuildEvent> {

    private GuildSettingsManager guildSettingsManager;

    public GuildJoinLeaveListener() {
        super(GuildCreateEvent.class, GuildDeleteEvent.class);
    }

    @Override
    public void execute(GuildEvent event) {
        if (this.guildSettingsManager == null)
            this.guildSettingsManager = Sparky.getInstance().getManager(GuildSettingsManager.class);

        if (event instanceof GuildCreateEvent) {
            this.executeCreate((GuildCreateEvent) event);
        } else if (event instanceof GuildDeleteEvent) {
            this.executeDelete((GuildDeleteEvent) event);
        }
    }

    private void executeCreate(GuildCreateEvent event) {
        this.guildSettingsManager.loadGuildSettings(event.getGuild().getId());
    }

    private void executeDelete(GuildDeleteEvent event) {
        this.guildSettingsManager.unloadGuildSettings(event.getGuildId());
    }

}

File: src/main/kotlin/dev/esophose/discordbot/listener/GuildJoinLeaveListener.kt
package dev.esophose.discordbot.listener

import dev.esophose.discordbot.Sparky
import dev.esophose.discordbot.manager.GuildSettingsManager
import discord4j.core.event.domain.guild.GuildCreateEvent
import discord4j.core.event.domain.guild.GuildDeleteEvent
import discord4j.core.event.domain.guild.GuildEvent

class GuildJoinLeaveListener : Listener<GuildEvent>(GuildCreateEvent::class, GuildDeleteEvent::class) {

    private var guildSettingsManager: GuildSettingsManager? = null

    override fun execute(event: GuildEvent) {
        if (this.guildSettingsManager == null)
            this.guildSettingsManager = Sparky.getManager(GuildSettingsManager::class)

        when (event) {
            is GuildCreateEvent -> executeCreate(event)
            is GuildDeleteEvent -> executeDelete(event)
        }
    }

    private fun executeCreate(event: GuildCreateEvent) {
        this.guildSettingsManager!!.loadGuildSettings(event.guild.id)
    }

    private fun executeDelete(event: GuildDeleteEvent) {
        this.guildSettingsManager!!.unloadGuildSettings(event.guildId)
    }

}